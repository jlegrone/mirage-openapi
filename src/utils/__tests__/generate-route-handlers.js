import SwaggerApi from 'sway'
import generateRouteHandlers from '../generate-route-handlers'
import definition from '../../__tests__/data/petstore.swagger-v2.json'

let handlerObjects
beforeAll(async () => {
  const api = await SwaggerApi.create({ definition })
  handlerObjects = generateRouteHandlers({ api })
})

describe('generate route handlers', () => {
  test('generates handler objects', () => {
    expect(handlerObjects).toHaveLength(14)
    expect(handlerObjects).toMatchSnapshot()
  })

  test('generates valid methods', () => {
    const validMethodRegex = /^(get|post|patch|put|delete|options|head)$/

    handlerObjects.forEach(({ handler }) => {
      Object.keys(handler).forEach((method) => {
        expect(method).toMatch(validMethodRegex)
      })
    })
  })

  test('generates valid paths', () => {
    handlerObjects.forEach(({ path }) => {
      expect(path).not.toMatch(/{(.*?)}/g)
    })
  })

  test('generates valid handlers', () => {
    handlerObjects.forEach(({ handler }) => {
      expect(handler).toBeTruthy()
    })
  })
})
