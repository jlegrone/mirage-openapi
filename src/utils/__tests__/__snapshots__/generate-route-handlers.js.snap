// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate route handlers generates handler objects 1`] = `
Array [
  Object {
    "handler": Object {
      "post": [Function],
      "put": [Function],
    },
    "path": "/pet",
  },
  Object {
    "handler": Object {
      "get": [Function],
    },
    "path": "/pet/findByStatus",
  },
  Object {
    "handler": Object {
      "get": [Function],
    },
    "path": "/pet/findByTags",
  },
  Object {
    "handler": Object {
      "delete": [Function],
      "get": [Function],
      "post": [Function],
    },
    "path": "/pet/:petId",
  },
  Object {
    "handler": Object {
      "post": [Function],
    },
    "path": "/pet/:petId/uploadImage",
  },
  Object {
    "handler": Object {
      "get": [Function],
    },
    "path": "/store/inventory",
  },
  Object {
    "handler": Object {
      "post": [Function],
    },
    "path": "/store/order",
  },
  Object {
    "handler": Object {
      "delete": [Function],
      "get": [Function],
    },
    "path": "/store/order/:orderId",
  },
  Object {
    "handler": Object {
      "post": [Function],
    },
    "path": "/user",
  },
  Object {
    "handler": Object {
      "post": [Function],
    },
    "path": "/user/createWithArray",
  },
  Object {
    "handler": Object {
      "post": [Function],
    },
    "path": "/user/createWithList",
  },
  Object {
    "handler": Object {
      "get": [Function],
    },
    "path": "/user/login",
  },
  Object {
    "handler": Object {
      "get": [Function],
    },
    "path": "/user/logout",
  },
  Object {
    "handler": Object {
      "delete": [Function],
      "get": [Function],
      "put": [Function],
    },
    "path": "/user/:username",
  },
]
`;
